docs/redis.md

```diff
@@ -12,10 +12,19 @@ To configure a receiver for your redis metrics, specify the following fields:
 | Field                 | Default                   | Description |
 | ---                   | ---                       | ---         |
 | `type`                | required                  | Must be `redis`. |
-| `address`            | `localhost:6379`          | The url exposed by redis |
+| `address`             | `localhost:6379`          | The url exposed by redis |
 | `collection_interval` | `60s`                     | A [time.Duration](https://pkg.go.dev/time#ParseDuration) value, such as `30s` or `5m`. |
 | `password`            |                           | The password used to connect to the server. |
 
+optional fields are: 
+| Field                 | Default                   | Description |
+| ---                   | ---                       | ---         |
+| `transport` | `tcp`| Defines the network to use for connecting to the server. |
+| `insecure` | `true` | whether to disable client transport security for the exporter's connection. |
```

Comment:
Can the value of config parameter insecure be inferred by the presence of the other 3 parameters: CertFilePath, CaFilePath, KeyFilePath? If so, we could cut one redundant parameter.

Reply:
good point. set that up.

Reply:
Do we know whether both of these cases are valid at otel level?

* insecure == true and none of the other 3 parameters are set
* insecure == false and none of the other 3 parameters are set

I was under the impression that `insecure == false and none of the other 3 parameters are set` is not a valid case, thus I recommended omitting `insecure`. But Quentin pointed out a good point that an unspecified CA path might mean to use the system default CAs, so it could potentially be a valid case though.

@Mrod1598 Do you know or can you test it out whether that is something otel supports?

Reply:
In fact, it would be better if we can separate the TLS implementation for Redis to a follow up PR and take the time we need to get it right. There is some complexity to it regarding:

* What config parameters make the most sense for Redis users. e.g. Unlike some other apps, Redis out of the box does not come with TLS enabled and require explicit user actions to enable it.
* An unspecified CA path might mean to use the system default CAs. It's unclear yet whether otel supports this. It could affect config design.
* Some receivers in the future might need TLS support as well, so the TLS related config we invent today would likely be the convention moving forward.
* We might need some end to end tests for TLS (Does otel have anything related already?)

Reply:
CC @quentinmit for visibility as we chatted about this offline.

Reply:
Ok removed TLS for the time being. will make that PR when this gets merged.

Reply:
> * or Redis users. e.g. Unlike some other apps, Redis out of the box does not come with TLS enabled and require explicit user actions to enable it.
> * An unspecified CA path might mean to use the system default CAs. It's unclear yet whether otel supports this. It could affe

will still look into this and see what I can figure out.

Reply:
> I was under the impression that `insecure == false and none of the other 3 parameters are set` is not a valid case, thus I recommended omitting `insecure`. But Quentin pointed out a good point that an unspecified CA path might mean to use the system default CAs, so it could potentially be a valid case though.
> 

just tested it and `insecure == false and none of the other 3 parameters are set`  is a valid config.